engine/src/main/java/org/pentaho/di/job/JobMeta.java

```diff
@@ -1167,7 +1158,7 @@ public void loadXML( Node jobnode, String fname, Repository rep, IMetaStore meta
         Node entrynode = XMLHandler.getSubNodeByNr( entriesnode, "entry", i );
         // System.out.println("Reading entry:\n"+entrynode);
 
-        JobEntryCopy je = new JobEntryCopy( entrynode, getDatabases(), slaveServers, rep, metaStore );
+        JobEntryCopy je = new JobEntryCopy( entrynode, getDatabases(), getSlaveServers(), rep, metaStore );
```

Comment:
is this returning *all* the slave servers, or just the local ones? I suppose the same risk is true for getDatabases().

Reply:
This is returning all slave servers. I tried looking at the usage  of various JobEntryInterface implementation to see how they are using the list of databases and SlaveServers and only one I found was in JobEntryCheckDbConnections which was trying to find the database connection name in the list of databaseConnections that was passed in. Haven't found any usage of slave server list.
But looking at the fact that it is job entry, if the KJB  has a KTR which has the db connection, the question is can it refer to any DB connection or only that is defined in the parent KJB. I think it should be  all connections and slave servers not just local.